docs/Tutorials/Permissioning/Upgrade-Permissioning-Contract.md

```diff
@@ -53,58 +48,125 @@ yarn run build
 
 ### 3. Update environment variables
 
-!!! important
+If using a `.env` file to configure environment variables, then
+the file must be in the `permissioning-smart-contracts` directory.
 
-    This step is only required if upgrading from v1.1.0 or earlier release of the
-    permissioning contract to v2.0.0 or later.
+!!! tip
 
-1. If they exist, delete the following environment variables:
-    * `ACCOUNT_INGRESS_CONTRACT_ADDRESS`
-    * `NODE_INGRESS_CONTRACT_ADDRESS`.
+    You can use environment variables to retain existing contracts if required. For example:
 
-1. If they exist, rename the following environment variables:
+    * `RETAIN_ADMIN_CONTRACT=true` to retain the current admin list
+    * `RETAIN_NODE_RULES_CONTRACT=true` to retain the current Node rules contract
+    * `RETAIN_ACCOUNT_RULES_CONTRACT=true` to retain the current Account rules contract
+
+1. Legacy: If they exist, rename the following environment variables:
     * `PANTHEON_NODE_PERM_ACCOUNT` to `BESU_NODE_PERM_ACCOUNT`
     * `PANTHEON_NODE_PERM_KEY` to `BESU_NODE_PERM_KEY`
-    * `PANTHEON_NODE_PERM_ENDPOINT` to `BESU_NODE_PERM_ENDPOINT`.
+    * `PANTHEON_NODE_PERM_ENDPOINT` to `BESU_NODE_PERM_ENDPOINT`
 
-### 4. Deploy the contract
+2. If updating from v1 to v2, you need to re-deploy the `NodeIngress` contract. In order to do this, first delete the `NODE_INGRESS_CONTRACT_ADDRESS` environment variable.
```

Comment:
Suggestion: "If updating from v1 to v2, delete the `NODE_INGRESS_CONTRACT_ADDRESS` environment variable and re-deploy the `NodeIngress` contract."

Reply:
I had this split into two separate instructions, as the first part is under the heading "update environment variables" and the second part is under "deploy the contracts"